contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,365 @@
+        uint128 pcvDepositWeight = pcvDepositInfos[pcvDeposit].weight;
+        uint idealDepositBalance = pcvDepositWeight * totalBalance / totalWeight;
+        uint pcvDepositBalance = IPCVDeposit(pcvDeposit).balance();
+
+        if (pcvDepositBalance > idealDepositBalance) {
+            // PCV deposit balance is too high. Withdraw from it into the aggregator.
+            uint overage = pcvDepositBalance - idealDepositBalance;
+            IPCVDeposit(pcvDeposit).withdraw(address(this), overage);
+        } else if (pcvDepositBalance < idealDepositBalance) {
+            // PCV deposit balance is too low. Pull from the aggregator balance if we can.
+            uint defecit = idealDepositBalance - pcvDepositBalance;
+            if (IERC20(token).balanceOf(address(this)) >= defecit) {
+                IERC20(token).safeTransfer(pcvDeposit, defecit);
+                IPCVDeposit(pcvDeposit).deposit();
+            } else {
+                // Emit event so that we know to do a full rebalance
+            }
+        } else {
+            // PCV deposit balance is exactly where it needs to be. Don't touch it.
+        }
+    }
+
+
+    function _rebalance() internal {
+        // Get the balances of all pcvDepositInfos
+
+        uint totalBalance = getTotalBalance();
+
+        uint[] memory depositAmountsNeeded = new uint[](pcvDepositAddresses.length());
+
+        for (uint i=0; i<pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            uint pcvDepositBalance = IPCVDeposit(pcvDepositAddress).balance();
+            uint pcvDepositWeight = pcvDepositInfos[pcvDepositAddress].weight;
+
+            uint idealDepositBalance = pcvDepositWeight * totalBalance / totalWeight;
+
+            if (pcvDepositBalance > idealDepositBalance) {
+                // Has an overage. Let's take it.
+                uint overage = pcvDepositBalance - idealDepositBalance;
+                IPCVDeposit(pcvDepositAddress).withdraw(address(this), overage);
+            } else if (pcvDepositBalance < idealDepositBalance) {
+                // Needs a deposit. Let's write that down.
+                uint needed = idealDepositBalance - pcvDepositBalance;
+                depositAmountsNeeded[i] = needed;
+            } else {
+                // Do nothing
+                // This accounts for the rare = case, but is definitely necessary.
+            }
+        }
+
+        for (uint i=0; i<depositAmountsNeeded.length; i++) {
+            uint amountNeeded = depositAmountsNeeded[i];
+            if (amountNeeded == 0) continue;
+
+            address depositAddress = pcvDepositAddresses.at(i);
+
+            IERC20(token).safeTransfer(depositAddress, amountNeeded);
+            IPCVDeposit(depositAddress).deposit();
+        }
+    }
+
+    function _addPCVDeposit(address depositAddress, uint128 weight) internal {
+        if (pcvDepositAddresses.contains(depositAddress)) {
```

Comment:
You should call pcvDepositInfos[depositAddress] and seeing if that deposit has a weight instead of looping over the pcvDepositAddresses to see if there is a match.